listener.go

```diff
@@ -28,7 +28,7 @@ func NewListener(l net.Listener) *GracefulListener {
 type gracefulConn struct {
 	net.Conn
 	lastHTTPState http.ConnState
-	forceClosed   bool
+	protected     bool
```

Comment:
@jeremyschlatter this name does not seem to be relevant imho. If `protected` is set to `true`, it means that a request currently being executed on the connection won't be interrupted by shutdown, it is protected from interruption so to speak. On the other hand `calledStart` tells that some start was called but that does not clarify things enough imho.